### PR TITLE
use new `sim` object for units

### PR DESCRIPTION
**Pull request by psychocoderHPC**
follow-up of #5066

- UNIT_SPEED to sim.unit.speed()
- SPEED_OF_LIGHT to sim.pic.getSpeedOfLight()
- BASE_CHARGE to sim.pic.getBaseCharge()
- BASE_MASS to sim.pic.getBaseMass()
- ElectronCharge to sim.pic.getElectronCharge()
- ELECTRON_MASS to sim.pic.getElectronMass()

Old setups should not be affected by this change.

# Review

Check each PR separately, some of the renaming are performed by the small renaming tool.

- [x] rebase after #5090 is merged

### CODE REVIEW

**Review by chillenzer — DISMISSED** [review]
Looks fine from skimming through. Happy when the CI is, too.

File: include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/ArbitraryOrderFDTD.hpp [hunk 1]
```diff
@@ -68,13 +68,13 @@ namespace picongpu
                     auto const invCorrectedCell2Sum = invCellSizeSquaredSum * additionalFactor * additionalFactor;
                     auto const maxC_DT = 1.0_X / math::sqrt(invCorrectedCell2Sum);
                     constexpr auto dt = getTimeStep();
```
Comment (PrometheusPi) [inline include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/ArbitraryOrderFDTD.hpp:70]: why do we have both `getTimeStep()` and `sim.pic.getDt()` - shouldn't we just have one?
(@BrianMarre just said this is related to substepping)

**Comment by psychocoderHPC** [comment]
@chillenzer since I need to rebase this PR anyway we should merge #5090 because there will be more conflicts.

File: include/picongpu/param/speciesDefinition.param
Comment (PrometheusPi) [inline include/picongpu/param/speciesDefinition.param:1]: fine with me, but this change is out of scope

File: share/picongpu/benchmarks/TWEAC-FOM/include/picongpu/param/incidentField.param
Comment (PrometheusPi) [inline share/picongpu/benchmarks/TWEAC-FOM/include/picongpu/param/incidentField.param:1]: out of the scope of this pull request, but this file has a lot of code repetition that could be avoided.

File: share/picongpu/pypicongpu/template/include/picongpu/param/speciesDefinition.param.mustache [hunk 2]
```diff
@@ -39,10 +39,10 @@ namespace picongpu
 
         {{#constants}}
             {{#mass}}
-                value_identifier(float_X, MassRatio_{{{typename}}}, {{{mass_si}}} / SI::BASE_MASS_SI);
+                value_identifier(float_X, MassRatio_{{{typename}}}, {{{mass_si}}} / sim.si.getBaseMass());
```
Comment (PrometheusPi) [inline share/picongpu/pypicongpu/template/include/picongpu/param/speciesDefinition.param.mustache:42]: @BrianMarre Does this work in the mustache template?

File: share/picongpu/tests/Shadowgraphy/include/picongpu/param/incidentField.param [hunk 3]
```diff
@@ -37,11 +37,11 @@ namespace picongpu
             {
                 struct MyGaussianParam
                 {
-                    static constexpr float_64 WAVE_LENGTH_SI = 800e-9; // TIME_PERIOD_SI * SI::SPEED_OF_LIGHT_SI;
-                    static constexpr float_64 TIME_PERIOD_SI = WAVE_LENGTH_SI / SI::SPEED_OF_LIGHT_SI;
+                    static constexpr float_64 WAVE_LENGTH_SI = 800e-9; // TIME_PERIOD_SI * sim.si.getSpeedOfLight();
```
Comment (PrometheusPi) [inline share/picongpu/tests/Shadowgraphy/include/picongpu/param/incidentField.param:40]: Fine with me, but this was already commented out code

**Review by PrometheusPi — APPROVED** [review]
some minor remarks not effecting this pull request.

File: share/picongpu/pypicongpu/template/include/picongpu/param/speciesDefinition.param.mustache (hunk 2 above)
Comment (BrianMarre) [reply share/picongpu/pypicongpu/template/include/picongpu/param/speciesDefinition.param.mustache:42]: yes, mustache doesn't care about non substitution fields and so long as generates a valid set of param files, which is tested in the CI everything is fine.

File: include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/ArbitraryOrderFDTD.hpp (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/ArbitraryOrderFDTD.hpp:70]: ~~`getTimeStep()` can be removed in the future. ~~ Oh yes could be that it is for sub stepping, so it is independent of what the `sim` object is providing,

File: include/picongpu/param/speciesDefinition.param
Comment (psychocoderHPC) [reply include/picongpu/param/speciesDefinition.param:1]: Normally it did a lind replacement but here I saw that it makes no sense.

File: share/picongpu/benchmarks/TWEAC-FOM/include/picongpu/param/incidentField.param
Comment (psychocoderHPC) [reply share/picongpu/benchmarks/TWEAC-FOM/include/picongpu/param/incidentField.param:1]: yes, we need someone who is refactoring the field solver, boundary conditions and incident fields anyway. I am open for contributions^^

File: share/picongpu/tests/Shadowgraphy/include/picongpu/param/incidentField.param (hunk 3 above)
Comment (psychocoderHPC) [reply share/picongpu/tests/Shadowgraphy/include/picongpu/param/incidentField.param:40]: with auto replacement I can not take care, thats why I always say we should not have dead code in files. But maybe it is in a comment because in the past it was at this point in the param files not possible to access DELTA_T and SPEED_OF_LIGHT.
